src/crepe/system/CollisionSystem.cpp

```diff
@@ -1,5 +1,393 @@
+	else {
+		info.first_rigidbody.data.linear_velocity = {0,0};
+	}
+}
+
+std::vector<std::pair<CollisionSystem::CollisionInternal,CollisionSystem::CollisionInternal>> CollisionSystem::check_collisions(std::vector<collider_variant> & colliders) {
+	
+	
+	// TODO:
+	// If no colliders skip
+	// Check if colliders has rigidbody if not skip
+
+	// TODO:
+	// If amount is higer than lets say 16 for now use quadtree otwerwise skip
+	// Quadtree code
+	// Quadtree is placed over the input vector
+
+	std::vector<std::pair<CollisionInternal,CollisionInternal>> collisions_ret;
+	for (size_t i = 0; i < colliders.size(); ++i) {
+    std::visit([&](auto& inner_collider_ref) {
+			if (!inner_collider_ref.get().active) return;
+			auto inner_components = get_active_transform_and_rigidbody(inner_collider_ref.get().game_object_id);
+			if (!inner_components) return;
+			for (size_t j = i + 1; j < colliders.size(); ++j) {
+				std::visit([&](auto& outer_collider_ref) { 
+					if (!outer_collider_ref.get().active) return;
+					if (inner_collider_ref.get().game_object_id == outer_collider_ref.get().game_object_id) return;
+					auto outer_components = get_active_transform_and_rigidbody(outer_collider_ref.get().game_object_id);
+					if (!outer_components) return;
+					CollisionInternalType type = check_collider_type(colliders[i],colliders[j]);
+					if(!check_collision({
+															.collider = colliders[i],
+															.transform = inner_components->first,
+															.rigidbody = inner_components->second,
+															},
+															{
+															.collider = colliders[j],
+															.transform = outer_components->first,
+															.rigidbody = outer_components->second,
+															},
+															type)) return;
+					collisions_ret.emplace_back(
+                        CollisionInternal{colliders[i], inner_components->first.get(), inner_components->second.get()},
+                        CollisionInternal{colliders[j], outer_components->first.get(), outer_components->second.get()}
+                    );
+				}, colliders[j]);
+			}
+    }, colliders[i]);
+	}
+	
+	return collisions_ret;
+}
+
+std::optional<std::pair<std::reference_wrapper<Transform>, std::reference_wrapper<Rigidbody>>>
+CollisionSystem::get_active_transform_and_rigidbody(game_object_id_t game_object_id) {
+    RefVector<Transform> transforms = this->component_manager.get_components_by_id<Transform>(game_object_id);
+    if (transforms.empty()) return std::nullopt;
+
+    RefVector<Rigidbody> rigidbodies = this->component_manager.get_components_by_id<Rigidbody>(game_object_id);
+    if (rigidbodies.empty()) return std::nullopt;
+
+    Transform& transform = transforms.front().get();
+    if (!transform.active) return std::nullopt;
+
+    Rigidbody& rigidbody = rigidbodies.front().get();
+    if (!rigidbody.active) return std::nullopt;
+
+    // Return the active components
+    return std::make_pair(std::ref(transform), std::ref(rigidbody));
```

Comment:
Is er een verschil tussen `std::make_pair` gebruiken en gewoon `return {transform, rigidbody};` doen zoals je eerder gebruikte?

Reply:
Geen verschil. Ik heb het gezet naar std::make_pair. ik vind dat fijner geen andere goede reden.